File: 1572. Matrix Diagonal Sum/Solution.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0, n = mat.length;
        for (int i = 0; i < n; i++) {
            sum += mat[i][i] + mat[i][n - 1 - i];
        }
        return n % 2 == 1 ? sum - mat[n / 2][n / 2] : sum;
    }
}

File: 1572. Matrix Diagonal Sum/Solution.py
from typing import List


class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        res = sum(mat[i][i] + mat[i][n - 1 - i] for i in range(n))
        return res - mat[n // 2][n // 2] if n % 2 else res
